public/app/plugins/datasource/loki/language_provider.ts

```diff
@@ -439,7 +441,8 @@ export default class LokiLanguageProvider extends LanguageProvider {
   }
 
   async fetchLabelValues(key: string): Promise<string[]> {
-    const interpolatedKey = this.datasource.interpolateString(key);
+    const interpolatedKey = encodeURIComponent(this.datasource.interpolateString(key));
```

Comment:
hmm.. i agree this fixes the problem, but i'm not sure if this is the way to go.

i think valid Loki label names are the same as valid Prometheus label names and those are

> Label names may contain ASCII letters, numbers, as well as underscores. They must match the regex [a-zA-Z_][a-zA-Z0-9_]*.

so there is nothing there that needs escaping.

or, in other words: if it is not a valid label-name, it does not exist in Loki. so there will be no label-values for it.

so i think we can check that label-name if it is valid, and if it is not, we can just return an empty-array. 

what do you think?

or, in a

Reply:
Yep, adding encoding here will not fix the root cause. Even more, by just applying the fix here, the "autocomplete" view is still showing containing some false labels.
That's why I also added those lines: https://github.com/grafana/grafana/pull/49887/files#diff-0d43c145c3bca230cf6677772375619afc5f3ae4581ff294dfc9582e5c5970c4L279-R282

I think the fix in `getLabelCompletionItems` is in the right place, because now, `labelKeys` with specialcharacters won't even passed to the `fetchLabelValues` function. And only `labelKeys` conforming will be passed to `fetchLabelValues`. You are right, that if they conform, we do not need encoding. On the other hand it does not really hurt adding encoding here, or does it?